src/vroom_fct.cc

```diff
@@ -73,9 +73,9 @@ cpp11::integers read_fct_implicit(vroom_vec_info* info, bool include_na) {
           if (na_level == NA_INTEGER) {
             na_level = max_level++;
             levels.push_back(NA_STRING);
-            out[i++] = na_level;
             level_map[str.str()] = na_level;
```

Comment:
I feel like this may need to come out of the `if(na_level == NA_INTEGER)` also? It feels like we aren't adding additional NA strings to the `level_map`, after the first one.

Reply:
IIUC that won't affect correctness, but I guess it could make things slower. In any case, the intention of `level_map` certainly seems to be documenting everything we've seen so far on our "level discovery" journey.

Reply:
No you're right, it should definitely come out too. Thank you!